models/kernels/Pdm03.h

```diff
@@ -93,6 +94,152 @@ extern "C"
 
     }
 
+    void pdm03_wrapper(pdm03_struct* pdm_data)
+    {
+        return Pdm03(pdm_data->modelDay,
+        pdm_data->Cpar,
+        pdm_data->B,
+        &pdm_data->XHuz,
+        pdm_data->Huz,
+        &pdm_data->OV,
+        &pdm_data->AE,
+        &pdm_data->XCuz,
+        pdm_data->effPrecip,
+        pdm_data->PE,
+        pdm_data->Kv);
+    }
+
+    /*
+    %%=========================================================================
+    %% Hamon Potential ET model
+    %% INPUTS
+    %%   avgTemp      - average air temperature this time step (C)
+    %%   Latitude     - latitude of catchment centroid (degrees)
+    %%   day of year  - (1-365) or (1-366) in leap years
+    % OUTPUTS
+    %%   PE           - potential evapotranspiration this time step
+    %%=========================================================================
+    */
+
+    inline double calculateHamonPE(double avgTemp, double Latitude, int day_of_year)
+    {
+        double solarDeclination, dayLength, eStar, PE;
+
+        solarDeclination =
+            asin(0.39795*cos(0.2163108 + 2.0 * atan(0.9671396*tan(0.00860*
+                                                (double)(day_of_year-186)))));
+        dayLength = 24.0 - (24.0/M_PI)*(acos((sin(0.8333*M_PI/180.0)+
+                    sin(Latitude*M_PI/180.0)*sin(solarDeclination))/
+                    (cos(Latitude*M_PI/180.0)*cos(solarDeclination))));
+        eStar = 0.6108*exp((17.27*avgTemp)/(237.3+avgTemp));
+        PE = (715.5*dayLength/24.0)*eStar/(avgTemp + 273.2);
+
+        return PE;
+    }
+
+     /*
+     * convert Gregorian days to Julian date
+     *
+     * Modify as needed for your application.
+     *
+     * The Julian day starts at noon of the Gregorian day and extends
+     * to noon the next Gregorian day.
+     *
+     */
+    /*
+    ** Takes a date, and returns a Julian day. A Julian day is the number of
+    ** days since some base date  (in the very distant past).
+    ** Handy for getting date of x number of days after a given Julian date
+    ** (use jdate to get that from the Gregorian date).
+    ** Author: Robert G. Tantzen, translator: Nat Howard
+    ** Translated from the algol original in Collected Algorithms of CACM
+    ** (This and jdate are algorithm 199).
+    */
+
+
+    inline double greg_2_jul(long year, long mon, long day, long h, long mi, double se)
```

Comment:
If we keep these around long enough, a date/time utility header might be a good idea.